src/l1_fetcher.rs

```diff
@@ -34,6 +36,49 @@ pub enum L1FetchError {
     GetTx,
 }
 
+#[derive(Default)]
+struct L1Metrics {
+    // Metrics variables.
+    l1_blocks_processed: u64,
+    l2_blocks_processed: u64,
+    latest_l1_block_nbr: u64,
+    latest_l2_block_nbr: u64,
+
+    first_l1_block: Option<u64>,
+    last_l1_block: Option<u64>,
+}
+
+impl L1Metrics {
+    fn print(&mut self) {
+        if self.first_l1_block.is_none()
+            || self.latest_l1_block_nbr == 0
+            || self.latest_l2_block_nbr == 0
+        {
+            return;
+        }
+
+        let first_l1_block = self.first_l1_block.unwrap();
+        let progress = match self.last_l1_block {
+            Some(last_l1_block) => {
+                let total = last_l1_block - first_l1_block;
+                let cur = self.latest_l1_block_nbr - first_l1_block;
+                let perc = (cur * 100) / total;
+                format!("{perc}%")
+            }
+            None => "∞".to_string(),
+        };
+
+        tracing::info!(
+            "PROGRESS: [{}] CUR BLOCK L1: {} L2: {} TOTAL BLOCKS PROCESSED L1: {} L2: {}",
```

Comment:
Nit: would this look better padded into columns? See this [doc](https://doc.rust-lang.org/std/fmt/), I'm thinking something like this:

```
PROGRESS: [  1%] CUR BLOCK L1: {} L2: {} TOTAL BLOCKS PROCESSED L1: {} L2: {}
PROGRESS: [ 50%] CUR BLOCK L1: {} L2: {} TOTAL BLOCKS PROCESSED L1: {} L2: {}
PROGRESS: [100%] CUR BLOCK L1: {} L2: {} TOTAL BLOCKS PROCESSED L1: {} L2: {}
```

Reply:
> would this look better padded into columns?

The way I see it, no. How would you tabulate it?

I quickly tested by reserving enough space for those block numbers to fit big values and it looks horrible:
```
2023-10-20T05:26:57.936783Z  INFO PROGRESS: [94%] CUR BLOCK L1:             16830084 L2:                  272 TOTAL BLOCKS PROCESSED L1:                 5858 L2:                    4
```

I'd rather keep it unpadded.

Reply:
Like this:
```
2023-10-20T07:52:01.438590Z  INFO Loading previous state file...
2023-10-20T07:52:21.459177Z  INFO PROGRESS: [ 5%] CUR BLOCK L1: 16836100 L2: 278 TOTAL BLOCKS PROCESSED L1: 3350 L2: 9
2023-10-20T07:52:31.460196Z  INFO PROGRESS: [10%] CUR BLOCK L1: 16841220 L2: 282 TOTAL BLOCKS PROCESSED L1: 8314 L2: 25
2023-10-20T07:52:41.461563Z  INFO PROGRESS: [15%] CUR BLOCK L1: 16845700 L2: 284 TOTAL BLOCKS PROCESSED L1: 11793 L2: 29
2023-10-20T07:52:51.463238Z  INFO PROGRESS: [16%] CUR BLOCK L1: 16847492 L2: 290 TOTAL BLOCKS PROCESSED L1: 15652 L2: 65
2023-10-20T07:53:01.464496Z  INFO PROGRESS: [19%] CUR BLOCK L1: 16849924 L2: 291 TOTAL BLOCKS PROCESSED L1: 16515 L2: 66
2023-10-20T07:53:11.465159Z  INFO PROGRESS: [20%] CUR BLOCK L1: 16851332 L2: 294 TOTAL BLOCKS PROCESSED L1: 20202 L2: 75
2023-10-20T07:53:21.466520Z  INFO PROGRESS: [22%] CUR BLOCK L1: 16852740 L2: 294 TOTAL BLOCKS PROCESSED L1: 20202 L2: 75
2023-10-20T07:53:31.467610Z  INFO PROGRESS: [24%] CUR BLOCK L1: 16855172 L2: 296 TOTAL BLOCKS PROCESSED L1: 24528 L2: 77
```
:question:

Reply:
Yes, that's looking nice! :+1: